app/javascript/packages/modal/modal-element.ts

```diff
@@ -1,32 +1,26 @@
-import { createFocusTrap } from 'focus-trap';
-import type { FocusTrap } from 'focus-trap';
-
 class ModalElement extends HTMLElement {
-  trap: FocusTrap;
-
   connectedCallback() {
-    this.trap = createFocusTrap(this, { escapeDeactivates: false });
     this.addEventListener('click', this.#handleDismiss);
   }
 
   /**
    * Shows the modal dialog.
    */
   show() {
-    this.removeAttribute('hidden');
-    this.classList.add('is-visible');
     this.ownerDocument.body.classList.add('usa-js-modal--active');
-    this.trap.activate();
+    this.#dialog.showModal();
   }
 
   /**
    * Hides the modal dialog.
    */
   hide() {
-    this.setAttribute('hidden', '');
-    this.classList.remove('is-visible');
     this.ownerDocument.body.classList.remove('usa-js-modal--active');
-    this.trap.deactivate();
+    this.#dialog.close();
+  }
+
+  get #dialog(): HTMLDialogElement {
+    return this.querySelector<HTMLDialogElement>('.modal__content')!;
```

Comment:
would it be clearer to specify the element? since the type signatures expect that?
```suggestion
    return this.querySelector<HTMLDialogElement>('dialog')!;
```

Reply:
Yeah I flip-flopped a bit on this one. The other nice thing about your suggestion is it means we can drop the generic type as well, since TypeScript can infer it from the argument.

```suggestion
    return this.querySelector('dialog')!;
```

I implemented it this way because I tend to avoid top-level element selectors in favor of something more specific wherever possible to avoid the possibility of selecting the wrong element, though it's rather doubtful in this case we'd have multiple dialog elements to need to worry about ambiguity.

Reply:
Updated in 094b42875.